tests/migration_test.go

```diff
@@ -932,23 +933,45 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 		Context("with an Fedora shared NFS PVC, cloud init and service account", func() {
 			var pvName string
 			var vmi *v1.VirtualMachineInstance
-			BeforeEach(func() {
+
+			initNFS := func(ipFamily k8sv1.IPFamily) {
 				tests.SkipNFSTestIfRunnigOnKindInfra()
 				pvName = "test-nfs" + rand.String(48)
 				// Prepare a NFS backed PV
 				By("Starting an NFS POD")
 				os := string(cd.ContainerDiskFedora)
-				nfsIP := tests.CreateNFSTargetPOD(os)
+				nfsIPs := tests.CreateNFSTargetPOD(os)
 				// create a new PV and PVC (PVs can't be reused)
 				By("create a new NFS PV and PVC")
+				nfsIP := libnet.GetIp(libnet.GetPodIpsStrings(nfsIPs), ipFamily)
+				Expect(nfsIP).NotTo(BeEmpty())
 				tests.CreateNFSPvAndPvc(pvName, "5Gi", nfsIP, os)
-			})
+			}
 
 			AfterEach(func() {
+				By("Deleting the VMI")
+				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
```

Comment:
AfterEach doesnt show errors, but in case there was a skip, the vmi wont be created, so its not accurate to use expect here i think

Reply:
Well, it's really bad. But from what I tested, aftereach is not invoked if the test is skipped.

Reply:
Well, it's really bad. But from what I tested. `AfterEach` is not invoked if the test is skipped.